### PR TITLE
add focus window method with some minor clang-format changes

### PR DESCRIPTION
**Pull request by weshoke**
When using Polyscope within Python on OSX, the window pops up behind the currently used windows and has to be focused in order to see it.  Add and use a `focusWindow` method so that it shows at the front when the windows is created.

* add a `focusWindow` method
* use `focusWindow` when a window is created

Also, there were some minor clang-format changes when editing these files.  I figured they should be included since the changes were made based on the config `.clang-format` file in this repository.

### CODE REVIEW

**Comment by nmwsharp** [comment]
Thanks for submitting! I'm 90% on board with this (esp. adding a `focusWindow` method).

I actually spent a while thinking about this a few years ago. It used to be that Polyscope would give focus when you call `init()`. However, that wasn't a good strategy. The problem is that many people write programs like 
```cpp
polyscope::init()
// some long-running computation
polyscope::show()
```

If we give focus at `init()`, then the user sees an unresponsive window during the long computation, and thinks Polyscope is frozen, so that's no okay. So then the alternative is to give focus on `show()`, as is effectively done in this PR. If I remember correctly, on some platforms this didn't work, because GLFW could not give a window focus after creation. It bothered me to have different behavior on different platforms, so I went with the current strategy. (I'm kicking myself now for not documenting exactly what platform setup this failed on.... I'd have to do some exploring to try to figure it out.)

At the very least, I'd certainly be on board with adding this functionality guarded behind an option. E.g. `options::giveFocusOnShow` or something, set initially to `false` to preserve the current behavior. As for enabling this behavior by default, let me at least poke around first and see if I can reproduce the situation that led to this behavior initially.

**Comment by weshoke** [comment]
Thanks for the feedback.  This makes sense.  I agree that options to enable it is a good idea.  There's already config to set options like "program_name" and "up_dir", so adding an extra option for this kind of behavior fits into that pattern.

**Comment by weshoke** [comment]
@nmwsharp I added the `giveFocusOnShow` option.  I can update docs and the Python bindings if this looks good.

File: src/polyscope.cpp [hunk 1]
```diff
@@ -156,6 +156,10 @@ void pushContext(std::function<void()> callbackFunction, bool drawDefaultUI) {
   // Make sure the window is visible
   render::engine->showWindow();
 
+  if (options::giveFocusOnShow) {
```
Comment (nmwsharp) [inline src/polyscope.cpp:159]: Could we move this to around L729 (just before the `pushContext()` in `show()`)?

Probably won't matter, but there are some other places `pushContext()` gets called in addition to inside `show()`, so putting the `focusWindow()` call actually inside `show()` will be more precise.

**Comment by nmwsharp** [comment]
Thanks for updating!

I left one small comment inline, otherwise looks good to me.

**Comment by weshoke** [comment]
@nmwsharp I've made the update.  Will make relevant changes to the Python bindings and docs

**Comment by weshoke** [comment]
https://github.com/nmwsharp/polyscope-py/pull/6

**Comment by nmwsharp** [comment]
awesome, thank you!

**Comment by nmwsharp** [comment]
FWIW on the docs front, don't worry about actually rebuilding them, or even getting the local server hooked up if it's being difficult; feel free to just make a copy-paste edit to the raw markdown. I appreciate you going the full distance on bindings and docs!

**Pull request merged**